cpp/merge_tools/apply_differential_update.cc

```diff
@@ -63,22 +64,42 @@ void CopyAndCollectPPNs(LocalDataDB * const local_data_db, MARC::Reader * const
                 record.truncate(first_local_field);
             }
 
-            previously_seen_ppns->emplace(PPN);
+            authority_ppns_in_delta->emplace(PPN);
             if (record.getFirstField("ORI") == record.end())
                 record.insertField("ORI", 'a', FileUtil::GetLastPathComponent(reader->getPath()));
-            writer->write(record);
+            if (check_input_ppns == false || authority_ppns_in_input.find(PPN) != authority_ppns_in_input.end())
+                writer->write(record);
         }
     }
 }
 
 
 void CopySelectedTypes(LocalDataDB * const local_data_db, const std::vector<std::string> &archive_members, MARC::Writer * const writer,
-                       const std::set<BSZUtil::ArchiveType> &selected_types, std::unordered_set<std::string> * const previously_seen_ppns)
+                       const std::set<BSZUtil::ArchiveType> &selected_types, std::unordered_set<std::string> * const authority_ppns_in_delta,
+                       const std::unordered_set<std::string>  &authority_ppns_in_input, bool check_input_ppns = false)
 {
     for (const auto &archive_member : archive_members) {
         if (selected_types.find(BSZUtil::GetArchiveType(archive_member)) != selected_types.cend()) {
             const auto reader(MARC::Reader::Factory(archive_member, MARC::FileType::BINARY));
-            CopyAndCollectPPNs(local_data_db, reader.get(), writer, previously_seen_ppns);
+            //only check if delta records are present in source record if it is a sekkor file, not if TA file etc.
+            check_input_ppns |= StringUtil::Contains(archive_member, "sekkor");
```

Comment:
The |= can cause side effects. Imagine you have an archive with multiple members, check_input_ppns is passed as false and then check_input_ppns would stay to true if the first member is sekkor but the others aren't which should not happen, but anyway it would be more consistant to use a copy here.

Reply:
Or it might even be better to move this to the calling site and decide it once depending on whether the archive file name contains sekkor